### ISSUE TITLE
cli.SwitchAttr assignment difference between assign to variable and dict?

### ISSUE THREAD

**Issue by stephenmm** [tomerfiliba/plumbum#264]
First of all I would like to thank you for the best cli tool EVER! Seriously great library!

I would like to be able to iterate over the switches but when I assign the cli.SwitchAttr to dict value it acts differently than it does when I assign it to a variable:

```
    args = {}
    desc = cli.SwitchAttr( ['-d', '--desc'], default="netlist --create", help="P4 submit description.")
    args['lib']  = cli.SwitchAttr(['--lib'],  default="my_sims",       help="Define the library to use.")
    args['view'] = cli.SwitchAttr(['--view'], default="config_sv",     help="Define the view to use.")       
    args['desc'] = "{0}".format( desc )
    strFrmts = {}
    def main( self ):
        for key, val in self.args.iteritems():
            self.strFrmts[key] = "{0}".format( val )
            print "key: " + key + " val: " + self.strFrmts[key]
        print "key: desc val: " + self.desc  
```

OUTPUT:
```
key: desc val: <plumbum.cli.switches.SwitchAttr object at 0x7f705b3a3590>
key: lib val: <plumbum.cli.switches.SwitchAttr object at 0x7f7448162610>                                                                                                                                                               
key: view val: <plumbum.cli.switches.SwitchAttr object at 0x7f7448162690>                                                                                                                                                              
key: desc val: netlist --create 
```

I tried looking through the code and although the code looks great I was not able to understand why there is a difference. 

Plumbum: 1.6.1
Python: 2.7.3
OS: rhel  6.5 (Santiago)

BTW As a side note to the main question I have also tried looking at some of the internals but I was not able to get the value of the switch. Here is a sample of what I was trying:

 ```
   def main( self ):
        for swinfo in self._switches_by_func.values(): # How to get all command line argument values?
            print "swinfo.names: {0}".format( swinfo.names )
            print "swinfo.argname: {0}".format( swinfo.argname )
            print "swinfo.argtype: {0}".format( swinfo.argtype )
            print "swinfo.list: {0}".format( swinfo.list )
            print "swinfo.func: {0}".format( swinfo.func )
            print "swinfo.func: {0}".format( swinfo.func )
            #print "swinfo.__get__(): {0}".format( swinfo.__get__() )
            pp.pprint( swinfo )   
```

**Comment by henryiii** [comment]
From a quick glance, I think you are missing the descriptors, part of Python's internal new style object syntax. The objects in question are python descriptors, meaning they define a `__get__` and `__set__` (or at least the first). So when you access the class member, you actually are doing `self.desc.__get__()`. This doesn't do anything special on the dict, however, so you no longer get the special descriptor access.

(This is how properties work, too, btw)

**Comment by henryiii** [comment]
Does it work if you replace `val` with `val.__get__(self, self.__class__)` ? (or whatever you named your Application class)

**Comment by stephenmm** [comment]
Holy crap! Nice work my friend... That worked. Now get back to working/moving/newborn stuff! :) Thanks for the quick feedback!

**Comment by stephenmm** [comment]
Sorry to be a bother but in order to get this working I had to define my switches like this:
`tmp    = cli.SwitchAttr(['--tmp'], help="Define the temporary build path.");args['tmp']    = tmp ` 

Which is pretty ugly and I tried to figure out why when I do it like I show above (assign to dict) the switch does not get "registered" with the application. I know you explained it above but I am still not sure how to implement this in a cleaner fashion. Don't get me wrong it works the way I have it but I am hoping by seeing a cleaner solution it might give me more insight into what is happening here.

Thanks for any help anyone can provide.
